### ISSUE TITLE
Sometimes, suppress_warnings misses one of its attributes

### ISSUE THREAD

**Issue by olebole** [scikit-image/scikit-image#2412]
skimage 0.12.3

When trying to compile/test skimage on Debian testing with numpy-1.12 RC, I sometimes (not always) get the following error, both with 2.7 and 3.5:
```
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/build/skimage-0.12.3/debian/tmp/usr/lib/python2.7/dist-packages/skimage/transform/tests/test_integral.py", line 46, in test_vectorized_integrate
    assert_equal(expected, integrate(s, r0, c0, r1, c1))  # test deprecated
  File "/build/skimage-0.12.3/debian/tmp/usr/lib/python2.7/dist-packages/skimage/transform/integral.py", line 86, in integrate
    warn("The syntax 'integrate(ii, r0, c0, r1, c1)' is "
  File "/build/skimage-0.12.3/debian/tmp/usr/lib/python2.7/dist-packages/skimage/_shared/_warnings.py", line 16, in warn
    warnings.warn(message, stacklevel=stacklevel)
  File "/usr/lib/python2.7/dist-packages/numpy/testing/utils.py", line 2199, in _showwarning
    self._orig_show(message, category, filename, lineno,
AttributeError: 'suppress_warnings' object has no attribute '_orig_show'
```
I assume this is a numpy problem, but I am not sure.

- [] Code example
- [] Relevant images (if any)
- [X] Operating system and version
- [X] Python version
- [X] scikit-image version (run `skimage.__version__`)

I already reported this for numpy: numpy/numpy#8413 I am however not sure whether this is a previously hidden problem on skimage. Please coordinate yourself ;-)

**Comment by olebole** [comment]
@sciunto Does the label imply that this is a problem of scikit-image and not of numpy? If so, could you put a comment on numpy/numpy#8413 please?

**Comment by sciunto** [comment]
@olebole No, I don't what causes that problem. There is obviously a bug somewhere but I'm not sure it's on the skimage side.

**Comment by sciunto** [comment]
@olebole Is it 0.12? Have you tried to compile with master? would be nice if you can try on mater.

**Comment by olebole** [comment]
@sciunto It is v0.12.3, probably [this tag](https://github.com/scikit-image/scikit-image/releases/tag/v0.12.3). If you have a specific commit in mind, this would be easier for me.

**Comment by sciunto** [comment]
In the mean time, I was digging in closed issues and in my memory. We had some race condition issues, but related to sphinx, not to the test suite (#2257). Sounds unrelated...

**Comment by sciunto** [comment]
@scikit-image/core Given https://github.com/numpy/numpy/issues/8413#issuecomment-269098759 we may want to do something before the next release (like ajust the num_thread to 1 to disable the feature). I tag for 0.13 then.

**Comment by seberg** [comment]
Maybe it would be possible to try to keep the actual asserts somehow non-parallel. So that only the stuff is parallel which is absolutely necessary to test what is supposed to be tested. We can fix this in numpy for 1.12, if you think this is ok in 1.11 (where the race condition probably exists too, but the effect may not be visible if you do not try to tests warnings precisely), then its probably fine.

**Comment by sciunto** [comment]
Fixed in numpy: https://github.com/numpy/numpy/issues/8413#issuecomment-271028303
